File: android/src/main/java/com/reactnativechimesdk/itf/SimpleRealTimeObserver.java
package com.reactnativechimesdk.itf;

import androidx.annotation.NonNull;

import com.amazonaws.services.chime.sdk.meetings.audiovideo.AttendeeInfo;
import com.amazonaws.services.chime.sdk.meetings.audiovideo.SignalUpdate;
import com.amazonaws.services.chime.sdk.meetings.audiovideo.VolumeUpdate;
import com.amazonaws.services.chime.sdk.meetings.realtime.RealtimeObserver;

public interface SimpleRealTimeObserver extends RealtimeObserver {

  String TAG = "SimpleRealTimeObserver";

  @Override
  default void onAttendeesDropped(@NonNull AttendeeInfo[] attendeeInfos) {

  }

  @Override
  default void onAttendeesJoined(@NonNull AttendeeInfo[] attendeeInfos) {
  }

  @Override
  default void onAttendeesLeft(@NonNull AttendeeInfo[] attendeeInfos) {
  }

  @Override
  default void onAttendeesMuted(@NonNull AttendeeInfo[] attendeeInfos) {
  }

  @Override
  default void onAttendeesUnmuted(@NonNull AttendeeInfo[] attendeeInfos) {
  }

  @Override
  default void onSignalStrengthChanged(@NonNull SignalUpdate[] signalUpdates) {
    // Stream.of(signalUpdates).forEach(it -> Log.d(TAG, "onSignalStrengthChanged " + it.getAttendeeInfo().getExternalUserId() + " - " + it.getSignalStrength().name()));
  }

  @Override
  default void onVolumeChanged(@NonNull VolumeUpdate[] volumeUpdates) {
    // Stream.of(volumeUpdates).forEach(it -> Log.d(TAG, "onVolumeChanged: "  + it.getAttendeeInfo().getExternalUserId() + " - " + it.getVolumeLevel()));
  }
}

File: android/src/main/java/com/reactnativechimesdk/itf/SimpleRealTimeObserver.kt
package com.reactnativechimesdk.itf

import android.util.Log
import com.amazonaws.services.chime.sdk.meetings.audiovideo.AttendeeInfo
import com.amazonaws.services.chime.sdk.meetings.audiovideo.SignalUpdate
import com.amazonaws.services.chime.sdk.meetings.audiovideo.VolumeUpdate
import com.amazonaws.services.chime.sdk.meetings.realtime.RealtimeObserver

interface SimpleRealTimeObserver : RealtimeObserver {

  companion object {
    const val TAG = "SimpleRealTimeObserver"
  }

  override fun onAttendeesDropped(attendeeInfo: Array<AttendeeInfo>) {
    Log.d(TAG, "onAttendeesDropped: ")
  }

  override fun onAttendeesJoined(attendeeInfo: Array<AttendeeInfo>) {
    Log.d(TAG, "onAttendeesJoined: ")
  }

  override fun onAttendeesLeft(attendeeInfo: Array<AttendeeInfo>) {
    Log.d(TAG, "onAttendeesLeft: ")
  }

  override fun onAttendeesMuted(attendeeInfo: Array<AttendeeInfo>) {
    Log.d(TAG, "onAttendeesMuted: ")
  }

  override fun onAttendeesUnmuted(attendeeInfo: Array<AttendeeInfo>) {
    Log.d(TAG, "onAttendeesUnmuted: ")
  }

  override fun onSignalStrengthChanged(signalUpdates: Array<SignalUpdate>) {
  }

  override fun onVolumeChanged(volumeUpdates: Array<VolumeUpdate>) {
  }
}
